app/models/course/material/folder.rb

```diff
@@ -15,6 +15,29 @@ class Course::Material::Folder < ActiveRecord::Base
   validate :validate_name_is_unique_among_materials
   validates_with FilenameValidator
 
+  # @!attribute [r] material_count
+  #   Returns the number of files in current folder.
+  calculated :material_count, (lambda do
+    Course::Material.select { count('*') }.
+      where { course_materials.folder_id == course_material_folders.id }
+  end)
+
+  # @!attribute [r] children_count
+  #   Returns the number of subfolders in current folder.
+  calculated :children_count, (lambda do
+    select('COUNT(*)').
```

Comment:
Is there a way to specify the FROM in squeal ?  it's not because the edge gem I think, the material_count one also cannot join.